### PR TITLE
Support NCCL 2.12 ~ 2.14

### PR DESCRIPTION
**Pull request by takagi**

### CODE REVIEW

File: .pfnci/schema.yaml [hunk 1]
```diff
@@ -109,6 +109,14 @@ nccl:
       "11.4":
       "11.5":
       "11.6":
+  "2.12":
+    spec: "2.12.*"
+    cuda:
+      "10.2":
+      "11.0":
+      "11.4":
+      "11.5":
```
Comment (kmaehashi) [inline .pfnci/schema.yaml:118]:
```suggestion
```

The release notes says: This NCCL release supports [CUDA 10.2](https://docs.nvidia.com/cuda/archive/10.2/index.html), [CUDA 11.0](https://docs.nvidia.com/cuda/archive/11.0/index.html), and [CUDA 11.6](https://docs.nvidia.com/cuda/archive/11.6/index.html).

https://docs.nvidia.com/deeplearning/nccl/release-notes/rel_2-12-7.html#rel_2-12-7

**Review by kmaehashi — CHANGES_REQUESTED** [review]

**Comment by kmaehashi** [comment]
The library installer also needs to be fixed for CUDA versions not supported by NCCL 2.12. 😅

**Comment by takagi** [comment]
Fixed!

**Comment by kmaehashi** [comment]
It seems a garbage file is in the commit 😅

**Comment by takagi** [comment]
oops...

**Comment by kmaehashi** [comment]
Sorry to comment sporadically :bow: could you enable NCCL 2.12 in CUDA 11.0? Otherwise we have no way to test NCCL 2.12 in CI.

**Comment by takagi** [comment]
ah, we can't test it in cuda116 yet...

**Review by kmaehashi — APPROVED** [review]

**Comment by kmaehashi** [comment]
cupyx.distributed test failing with timeout :thinking:
https://ci.preferred.jp/cupy.linux.cuda110.multi/97051/197863#L7888

/test multi

**Comment by kmaehashi** [comment]
Changing target to the next month release.

**Comment by takagi** [comment]
I failed to reproduce the timeout on my local.

**Comment by kmaehashi** [comment]
TODO: use NCCL 2.12 in CUDA 11.7 test (https://github.com/cupy/cupy/pull/6767#discussion_r896358629)

**Comment by takagi** [comment]
I hope it works...

**Comment by takagi** [comment]
No v2.14 yet. https://developer.download.nvidia.com/compute/redist/nccl/

**Comment by kmaehashi** [comment]
https://developer.download.nvidia.com/compute/redist/nccl/v2.14/nccl_2.14.3-1+cuda11.7_x86_64.txz

File: cupyx/tools/install_library.py [hunk 2]
```diff
@@ -153,36 +153,19 @@ def _make_nccl_record(
...
-    '11.4', '2.11.4', '2.11',
-    'nccl_2.11.4-1+cuda11.4_x86_64.txz'))
-_nccl_records.append(_make_nccl_record(
-    '11.3', '2.9.9', '2.9',
-    'nccl_2.9.9-1+cuda11.3_x86_64.txz'))
-_nccl_records.append(_make_nccl_record(
-    '11.2', '2.8.4', '2.8',
-    'nccl_2.8.4-1+cuda11.2_x86_64.txz'))
+    '11.x', '2.14.3', '2.14',  # CUDA 11.2+
+    'nccl_2.14.3-1+cuda11.4_x86_64.txz'))
```
Comment (kmaehashi) [inline cupyx/tools/install_library.py:159]:
```suggestion
    'nccl_2.14.3-1+cuda11.7_x86_64.txz'))
```

**Review by kmaehashi — APPROVED** [review]

**Pull request merged**